### ISSUE TITLE
"Failed to open" error message with DirectShow when trying to load HVEC video 5760 x 2880

### ISSUE THREAD

**Issue by ricvail** [RenderHeads/UnityPlugin-AVProVideo#1123]
**Describe the issue**
I'm trying to load a video file using DirectShow. **The file loads and plays normally using Win RT or Media Foundation** (with the codec installed), I tried installing the LAV filters and applying them in the "preferred filters" editor field, but it still doesn't work, This is the only error I get: 
![image](https://user-images.githubusercontent.com/25117864/156664455-481bcccb-b2d1-420c-8e61-999c5d228a8c.png)

**Your Setup (please complete the following information):**
 - Unity version: 2019.4.25f1
 - AVPro Video version: version 2.4.0 (plugin v.2.3.2f1-core), Core version
 - Operating system version: Windows 10 version 10.0.19043 Build 19043
 - Video specs (resolution, frame-rate, codec, file size): 
    -Resolution: 5760 x 2880
    -Codec: MPEG-H Part2/HEVC (H.265) (hvc1)
    -Framerate: 29.970030
    -File size: 19,5 MB
    -Other info: it's a 360 (non-stereo) video, downloaded from the web and played locally (referenced via absolute path)

**Videos**
https://player.vimeo.com/external/569129927.source.mp4?s=ff9baf5601ed9d939bcfb345eb13e427c341cb51&download=1

Regardless of the issue, I would like to know if there's any way to play a HVEC video with this quality on Windows without requiring the user to download and install third party software. We tried DirectShow because we hoped it would work out-of-the-box, but it doesn't seem to be the case.

**Comment by kahnivore** [comment]
Strange that that's the error you're getting - just to rule out anything going wrong with the absolute path string - have you tried loading this from streaming assets?

**Comment by ricvail** [comment]
I just tried loading the video from the streaming assets folder, both with the option "path->relative to streaming asset folder" and using a media reference, and I even tried loading it directly from the Vimeo url. In all three cases, the video loads and plays successfully with Media Foundation, but fails to load with Direct Show. I always get the error posted above, as soon as I press the "load" button, both in edit mode and in play mode.

**Comment by kahnivore** [comment]
Don't think LAV filters includes support for HEVC. Unfortunately the only way around this at the moment is with the Microsoft extension. We generally recommend using H264 in cases where wide usability is necessary, or prompting users to download the extension if the video fails to play (eg load and play a very basic HEVC video in background, if it fires an error event, assume this is due to lack of codec and prompt users to download).

**Comment by ricvail** [comment]
By the way you might want to change this bit in your documentation: 
![image](https://user-images.githubusercontent.com/25117864/156996191-d2f75c0a-091e-4bf3-833d-714f5fad9f93.png)

**Comment by kahnivore** [comment]
Sorry, my mistake, on second look - it does seem to inlclude HEVC 
![image](https://user-images.githubusercontent.com/44468274/156999531-c9b2294e-91ac-4da1-b731-a2d68b1943cc.png)

Are you sure you're setting the preferred filter correctly? Is your video a standard HEVC?

**Comment by ricvail** [comment]
This is how I set it up:
![image](https://user-images.githubusercontent.com/25117864/157002337-19a7893e-151c-4478-84d2-97f09fe49e79.png)
(I tried both with and without the audio filter)
I don't think there's anything unusual about our video, apart from it being 360 and relatively high res (5760 x 2880); but I don't know much about video codecs so I'm not sure what to look for... Our application downloads the video from the same link I provided in the first post, if you want to have a look at it 😅
How did you access that list in your screenshot? It was not included in my installation process, when I installed the LAV filters

Ps. I just noticed that I also get this warning message (I had hidden the warnings in the Unity console, sorry):
![image](https://user-images.githubusercontent.com/25117864/157004088-fb49fe20-c211-41b4-88c6-8f12f0fbd593.png)

**Comment by AndrewRH** [comment]
I just tried your video using the LAV Filters 0.76.1 and it worked fine...so i'm not sure what's wrong.
Perhaps your GPU isn't up to the task?  Or...perhaps you have another DirectShow codec pack installed that's conflicting with LAV...

H.265/HEVC isn't easy to support without installing a 3rd party codec (like LAV or Media Foundation's HEVC decoder), so H.264 may be your best bet.

**Comment by ricvail** [comment]
I have a GTX 1660 super... It's not a beast but it should be good enough, the GPU shouldn't be a problem, unless DirectShow is much less efficient than Media Foundation. I don't know about other codec packs (apart from Media Foundation's HEVC decoder) but maybe they were included in other programs I had installed before.
Either way, even if I got it to work with Direct Show it would still require end-users to install additional software to see the videos, so I might as well stick to Media Foundation.
If you want to look further into the issue I'm happy to help, but as far as I'm concerned we can close the ticket... Thanks for the support :)
